src/libasr/pass/pass_utils.h

```diff
@@ -346,10 +349,9 @@ namespace LCompilers {
                     }
 
                     if (fill_function_dependencies && ASR::is_a<ASR::ExternalSymbol_t>(*x.m_name)) {
-                        ASR::Function_t *f = ASR::down_cast2<ASR::Function_t>(current_scope->asr_owner);
                         ASR::ExternalSymbol_t* external_symbol = ASR::down_cast<ASR::ExternalSymbol_t>(x.m_name);
 
-                        if (f->m_symtab->parent == external_symbol->m_parent_symtab) {
+                        if (current_scope->get_counter() != external_symbol->m_parent_symtab->get_counter()) {
```

Comment:
@czgdp1807 This check will check if the dependency is not defined in the current scope. 

But, in few test cases, dependency might be nested somewhere inside the current_scope (dependency can be grandchild of current scope) in that case both counters will be different and it will add to the dependencies and will fail in asr.